lib/trino-parquet/src/test/java/io/trino/parquet/crypto/TestParquetEncryption.java

```diff
@@ -0,0 +1,1057 @@
+import static org.apache.parquet.hadoop.ParquetFileWriter.Mode.OVERWRITE;
+import static org.apache.parquet.hadoop.metadata.CompressionCodecName.SNAPPY;
+import static org.apache.parquet.hadoop.metadata.CompressionCodecName.UNCOMPRESSED;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.joda.time.DateTimeZone.UTC;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+
+/**
+ * Unit‑tests exercising Trino’s PME path.
+ */
+// ExampleParquetWriter is not thread-safe
+@TestInstance(PER_CLASS)
+@Execution(ExecutionMode.SAME_THREAD)
+public class TestParquetEncryption
+{
+    // ──────────────────────────────────  CONSTANTS  ────────────────────────────────────
+    private static final ColumnPath AGE_PATH = ColumnPath.fromDotString("age");
+    private static final ColumnPath ID_PATH = ColumnPath.fromDotString("id");
+
+    private static final byte[] KEY_AGE = "colKeyIs16ByteA?".getBytes(UTF_8);
+    private static final byte[] KEY_ID = "colKeyIs16ByteB?".getBytes(UTF_8);
+    private static final byte[] KEY_FOOT = "footKeyIs16Byte?".getBytes(UTF_8);
+
+    // one‑column schema
+    private static final MessageType AGE_SCHEMA =
+            MessageTypeParser.parseMessageType("message doc { required int32 age; }");
+
+    // two‑column schema
+    private static final MessageType TWO_COL_SCHEMA = MessageTypeParser.parseMessageType(
+            "message doc { required int32 age; required int32 id; }");
+
+    // ────────────────────────────────  PARAMETERISED TESTS  ─────────────────────────────
+
+    /**
+     * Column encryption only – footer left in plaintext.
+     */
+    @ParameterizedTest(name = "columnOnly_footerIntegrity={0}_compressed={1}")
+    @CsvSource({
+            "false,false",
+            "false,true",
+            "true,false",
+            "true,true"
+    })
+    void columnOnlyFooterPlaintext(boolean checkFooterIntegrity, boolean compressed)
+            throws IOException
+    {
+        File file = Files.createTempFile("pme‑col‑only", ".parquet").toFile();
+        file.deleteOnExit();
+
+        writeSingleColumnFile(file, compressed, /*encryptFooter*/ false);
+
+        List<Integer> values = readSingleColumnFile(
+                file,
+                new TestingKeyRetriever(checkFooterIntegrity ? Optional.of(KEY_FOOT) : Optional.empty(), Optional.of(KEY_AGE), Optional.empty()),
+                checkFooterIntegrity,
+                Optional.empty());
+
+        verifySequence(values, 100);
+    }
+
+    /**
+     * Column+ footer encryption (same column as above).
+     */
+    @ParameterizedTest(name = "column+footer_compressed={0}")
+    @CsvSource({"false", "true"})
+    void columnAndFooter(boolean compressed)
+            throws IOException
+    {
+        File file = Files.createTempFile("pme‑col‑foot", ".parquet").toFile();
+        file.deleteOnExit();
```

Comment:
Do we need to add the `deleteOnExit` everywhere ? Most of our existing tests with `createTempFile` don't bother